pkg/controllers/certificate/cert_rotation_controller.go

```diff
@@ -165,7 +165,7 @@ func (c *CertRotationController) syncCertRotation(secret *corev1.Secret) error {
 
 	var newCertData []byte
 	klog.V(1).Infof("Waiting for the client certificate to be issued")
-	err = wait.Poll(1*time.Second, 5*time.Minute, func() (done bool, err error) {
+	err = wait.PollUntilContextTimeout(context.TODO(), 1*time.Second, 5*time.Minute, false, func(ctx context.Context) (done bool, err error) {
```

Comment:
Theses are not equivalent, I think you meant to use `PollWithContextTimeout` as the docs said to do, however this does not exists.

Instead you can do:
```suggestion
  ctx, cancel := context.WithTimeout(context.TODO(), 5*time.Minute)
  defer cancel()
	err = wait.PollWithContext(ctx, 1*time.Second, 5*time.Minute, func(ctx context.Context) (done bool, err error) {
```

Reply:
Thanks for your collaboration @Jorropo, but unfortunately the `PollWithContext` method is also deprecated.

Reference: https://pkg.go.dev/k8s.io/apimachinery@v0.27.3/pkg/util/wait#PollWithContext

Reply:
Ah right mb.